File: 프로그래머스/lv0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.cpp
#include <string>
#include <vector>

// 머씩이 들고있는거 balls 나눠줄거 share balls의 구슬중 share을 고르는 경우의 수
// 조합?

using namespace std;
__int128 factorial(int a){
    if(a==1||a==0) return 1;
    return a*factorial(a-1);
}
int solution(int balls, int share) {
    int answer = 0;
    answer = factorial(balls) / (factorial(balls - share) * factorial(share));
    
    return answer;
}

File: 프로그래머스/lv0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.py
def solution(balls, share):

    a=1
    b=1
    c=0
    for i in range(balls, 0 , -1) :
        a = a * i
        c +=1
        if c == share :
            break
    for i in range(share, 0, -1) :
        b = b * i




    answer = a//b
    return answer